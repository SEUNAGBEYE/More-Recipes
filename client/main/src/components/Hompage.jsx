import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { allRecipes, recipeCategories, popularRecipes } from '../actions/Recipes';
import RecipeCircle from './recipes/RecipeCircle';


/**
 * @export
 * @class Homepage
 * @extends {Component}
 */
export class Homepage extends Component {

  /**
   * @description - Get Component Data
   *
   * @method componentDidMount
   *
   * @returns {void} void
   * @memberof Homepage
   */
  componentDidMount() {
    this.props.popularRecipes(3);
    this.props.recipeCategories();
    this.props.allRecipes(1, 3);
  }


  /**
   * @description - Render react component
   *
   * @method render
   *
   * @returns {Jsx} Jsx
   * @memberof Homepage
   */
  render() {
    return (
      <div>
        <main>
          <section className="banner">
            <div className="overlay">
              <h3 className="overlay__h3">
              Welcome To Recipes. All About Reciping
              </h3>
            </div>
          </section>

          <section>
            <div className="container">

              <h3 className="title"
                style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}>
                <Link to="/recipes" className="title__link">Recipes</Link>
              </h3>

              <div className="row">
                {
                  this.props.recipes.map((recipe) =>
                    <RecipeCircle key={recipe.id} recipe={recipe} />)
                }
              </div>

              <h3 style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}>
                <Link to="#" className="title-link">
                Categories</Link>
              </h3>

              <div className="row">
                {
                  this.props.categories
                    .map((category) =>
                      (<RecipeCircle key={category.id} recipe={category}
                        to={`/categories/${category.name}`} />))
                }
              </div>
              <h3 style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}>
                <Link to="#" className="title-link">Popular</Link>
              </h3>

              <div className="row">
                {
                  this.props.popularRecipe
                    .map((recipe) => (<RecipeCircle
                      key={recipe.id} recipe={recipe} />))
                }
              </div>

            </div>
          </section>

        </main>
      </div>
    );
  }
}

const propTypes = {
  recipes: PropTypes.array.isRequired,
  recipeCategories: PropTypes.func.isRequired,
  allRecipes: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
  popularRecipes: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

Homepage.propTypes = propTypes;

/**
 * mapStateToProps
 * @export
 * @param {Object} state
 *
 * @returns {Object} object
 */
export const mapStateToProps = (state) => ({
  recipes: state.recipes.allRecipes,
  popularRecipe: state.recipes.popularRecipes,
  categories: state.recipes.recipeCategories
});

export default connect(mapStateToProps, {
  allRecipes,
  recipeCategories,
  popularRecipes
})(Homepage);
