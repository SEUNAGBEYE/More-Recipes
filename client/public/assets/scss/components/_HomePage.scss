
.banner {
	height: 500px;
    background-image: url('../img/pasta1.jpeg');
    background-repeat: no-repeat;
    background-size: cover;

    /*Used to make it fixed when scrolling*/

    background-attachment: fixed;  
    position: relative;
}

/*Rules style alphabeticaly and to html mackup*/

.overlay {
	position: absolute;
	width: 100%;
	height: 100%;
	background: rgba(0, 0, 0,0.1);
	color: #FFF;
	display: flex;
    justify-content: center;
    align-items: center;

}

.overlay h3 {
	font-size: 35px;
    letter-spacing: 2px;
    padding: 12px
}

.overlay__h3{
  color: $white
}