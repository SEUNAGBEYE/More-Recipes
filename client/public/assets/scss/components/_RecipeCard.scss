.card{
  height: 100%;
  margin: 0 10px 0 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
}

.card-img-top{
  height: 150px;
}

.card-title{
color: $orange;
margin: 10px;
}

.card-text{
  height: 80px;
  color: black;
  margin: 0 0 25px 0;
}

.my-card{
  margin-bottom: 15px;
}
