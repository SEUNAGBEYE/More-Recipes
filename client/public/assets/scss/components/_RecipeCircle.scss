.box {
	/*background: grey;*/
	height: 250px;
	width: 100%;
	display: flex;
	justify-content: center;
	align-items: center;

}

.circle {
	height: 200px;
	width: 200px;
	border-radius: 100%;
	border: 2px solid #DDD;
	background: #EEE;
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: 20px;
}