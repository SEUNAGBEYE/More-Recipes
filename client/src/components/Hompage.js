import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import Navbar from './common/Navbar';
import Footer from './common/Footer';
import Recipe from '../actions/Recipes';

import {
  cappuccino,
  breakfast,
  cookies,
  desserts,
  dinner,
  dougnuts,
  drinks,
  frozen_desserts,
  jollof_rice,
  lunch,
  noodles,
  pasta,
  pastries,
  pies,
  chips,
  pizza,
  pudding,
  roasted_chicken,
  salad,
  seun,
  banana_split
} from '../helpers/Images';


/**
 * @class Homepage
 * @extends Component
 */
export default class Homepage extends Component {

  componentDidMount(){
    Recipe.allRecipes();
  }

  /**
   * returns {obeject} object
   * @memberOf Homepage
   */

  render(){
    return (
      <div>
        <Navbar />
        <main>
          <section className="banner">
            <div className="overlay">
              <h3 className='overlay__h3'> Welcome To Recipes. All About Reciping</h3>
            </div>	
          </section>

          <section>
            <div className="container">

              <h3 className='title' style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}><Link to="/my_recipes" className="title__link">Recipes</Link></h3>
              
              <div className="row">

                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <Link to="recipe_details">
                  
                    <div className="box">
                      <div className="circle"><img className ="circle" src= {pasta} /></div>
                    </div>
                    Pasta
                  </Link>

                </div>

                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <Link to="my_recipes">
                  
                    <div className="box">
                      <div className="circle"><img className ="circle" src={ banana_split } /></div>
                    </div>
                    Banana Split
                  </Link>

                </div>

                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="recipe_detail.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={salad} /></div>
                    </div>
                    Creme Salad
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                  
                    <div className="box">
                      <div className="circle"><img className ="circle" src={jollof_rice} /></div>
                    </div>
                    Jollof Rice
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={roasted_chicken} /></div>
                    </div>
                    Roasted chicken
                  </a>

                </div>
                <div className="col-sm-6 col-md-4" style={{fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    <div className="box">
                      <div className="circle"><img className ="circle" src={chips} /></div>
                    </div>
                    Madrelle Chips
                  </a>

                </div>
              </div>

              <h3 style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}><a href="recipes.html" className="title-link">Categories</a></h3>
              
              <div className="row">

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                   
                    <div className="box">
                      <div className="circle"><img className ="circle" src={breakfast} /></div>
                    </div>
                    Breakfast
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={lunch} /></div>
                    </div>
                    Lunch
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={dinner} /></div>
                    </div>
                    Dinner
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={desserts} /></div>
                    </div>
                    Desserts
                  </a> 

                </div>

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    <div className="box">
                      <div className="circle"><img className ="circle" src={noodles} /></div>
                    </div>
                    Noodles
                  </a>

                </div>
                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="category.html">
                    <div className="box">
                      <div className="circle"><img className ="circle" src={drinks} /></div>
                    </div>
                    Drinks
                  </a>

                </div>
              </div>

              <h3 style={{textAlign: 'center', marginTop: 40, fontSize: 32}}><a href="#" className="title-link">Popular</a></h3>

              <div className="row">

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="recipe_detail.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={jollof_rice} /></div>
                    </div>
                    Jollof Rice
                  </a>

                </div>


                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center'}}>
                  <a href="recipe_detail.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={cappuccino} /></div>
                    </div>
                    Cappuccino
                  </a>

                </div>

                <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center', marginBottom: 20}}>
                  <a href="recipe_detail.html">
                    
                    <div className="box">
                      <div className="circle"><img className ="circle" src={pizza} /></div>
                    </div>
                    Margherita Pizza
                  </a>

                </div>
              </div>

            </div>
          </section>
          
        </main>
      <Footer />
      </div>
    )
  }
}