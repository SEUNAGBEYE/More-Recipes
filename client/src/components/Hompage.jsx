import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { allRecipes, recipeCategories, popularRecipes } from '../actions/Recipes';
import RecipeCircle from './recipes/RecipeCircle';


/**
 * @class Homepage
 * @extends Component
 */
class Homepage extends Component {
  /**@returns {void} void
   * @memberof Homepage
   */
  componentDidMount() {
    this.props.popularRecipes(3);
    this.props.recipeCategories();
    this.props.allRecipes(1, 3);
  }


  /**
   *@returns {jsx} jsx
   * @memberof Homepage
   */
  render() {
    return (
      <div>
        <main>
          <section className="banner">
            <div className="overlay">
              <h3 className="overlay__h3"> Welcome To Recipes. All About Reciping</h3>
            </div>
          </section>

          <section>
            <div className="container">

              <h3 className="title" style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}><Link to="/recipes" className="title__link">Recipes</Link></h3>

              <div className="row">
                {
                  this.props.recipes.map((recipe) => <RecipeCircle key={recipe.id} recipe={recipe} />)
                }
              </div>

              <h3 style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}><a href="recipes.html" className="title-link">Categories</a></h3>

              <div className="row">
                {
                  this.props.categories.map((recipe) => <RecipeCircle key={recipe.id} recipe={recipe} />)
                }
              </div>
              <h3 style={{ textAlign: 'center', marginTop: 40, fontSize: 32 }}><a href="#" className="title-link">Popular</a></h3>

              <div className="row">
                {
                  this.props.popularRecipe.map((recipe) => <RecipeCircle key={recipe.id} recipe={recipe} />)
                }
              </div>

            </div>
          </section>

        </main>
      </div>
    );
  }
}

/**
 * mapStateToProps
 * @param {any} state
 * @returns {object} object
 */
const mapStateToProps = (state) => ({
  recipes: state.recipes.allRecipes,
  popularRecipe: state.recipes.popularRecipes,
  categories: state.recipes.recipeCategories
});

export default connect(mapStateToProps, { allRecipes, recipeCategories, popularRecipes })(Homepage);
