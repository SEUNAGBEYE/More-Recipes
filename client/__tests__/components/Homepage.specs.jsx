
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme';
import { Homepage, mapStateToProps } from '../../main/src/components/Hompage';

const state = {
  recipes: {
    recipeCategories: [{
      id: 1,
      name: 'Breakfast'
    }],
    popularRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
    allRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
  }
};


const props = {
  recipes: [],
  popularRecipe: [],
  popularRecipes: jest.fn(),
  allRecipes: jest.fn(),
  recipeCategories: jest.fn(),
  categories: [],
  loaded: true,
  history: {
    push: (() => {})
  },
  categories: [],
  auth: {
    isAuthenticated: true,
    user: {
      firstName: 'Seun',
      lastName: 'Agbeye'
    }
  }
};

const event = {
  preventDefault: jest.fn(),
  target: {
    search: {
      value: '',
    },
    id: ''
  }
}

describe('# Hompage', () => {
  const wrapper = shallow(
      <Homepage {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper.instance().props).toEqual(props)
    expect(wrapper).toMatchSnapshot()
    done();
  });

  it('should call mapStateToProps(state)', (done) => {
    mapStateToProps(state)
    done();
  }); 
});
