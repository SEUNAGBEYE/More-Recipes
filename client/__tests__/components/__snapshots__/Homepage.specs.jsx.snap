// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Hompage should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage
    allRecipes={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            3,
          ],
        ],
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "firstName": "Seun",
          "lastName": "Agbeye",
        },
      }
    }
    categories={Array []}
    history={
      Object {
        "push": [Function],
      }
    }
    popularRecipe={Array []}
    popularRecipes={
      [MockFunction] {
        "calls": Array [
          Array [
            3,
          ],
        ],
      }
    }
    recipeCategories={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    recipes={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <section
          className="banner"
        >
          <div
            className="overlay"
          >
            <h3
              className="overlay__h3"
            >
              Welcome To Recipes. All About Reciping
            </h3>
          </div>
        </section>
        <section>
          <div
            className="container"
          >
            <h3
              className="title"
              style={
                Object {
                  "fontSize": 32,
                  "marginTop": 40,
                  "textAlign": "center",
                }
              }
            >
              <Link
                className="title__link"
                replace={false}
                to="/recipes"
              >
                Recipes
              </Link>
            </h3>
            <div
              className="row"
            />
            <h3
              style={
                Object {
                  "fontSize": 32,
                  "marginTop": 40,
                  "textAlign": "center",
                }
              }
            >
              <Link
                className="title-link"
                replace={false}
                to="#"
              >
                Categories
              </Link>
            </h3>
            <div
              className="row"
            />
            <h3
              style={
                Object {
                  "fontSize": 32,
                  "marginTop": 40,
                  "textAlign": "center",
                }
              }
            >
              <Link
                className="title-link"
                replace={false}
                to="#"
              >
                Popular
              </Link>
            </h3>
            <div
              className="row"
            />
          </div>
        </section>
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="banner"
          >
            <div
              className="overlay"
            >
              <h3
                className="overlay__h3"
              >
                Welcome To Recipes. All About Reciping
              </h3>
            </div>
          </section>,
          <section>
            <div
              className="container"
            >
              <h3
                className="title"
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title__link"
                  replace={false}
                  to="/recipes"
                >
                  Recipes
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Categories
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Popular
                </Link>
              </h3>
              <div
                className="row"
              />
            </div>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="overlay"
            >
              <h3
                className="overlay__h3"
              >
                Welcome To Recipes. All About Reciping
              </h3>
            </div>,
            "className": "banner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3
                className="overlay__h3"
              >
                Welcome To Recipes. All About Reciping
              </h3>,
              "className": "overlay",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome To Recipes. All About Reciping",
                "className": "overlay__h3",
              },
              "ref": null,
              "rendered": "Welcome To Recipes. All About Reciping",
              "type": "h3",
            },
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <h3
                className="title"
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title__link"
                  replace={false}
                  to="/recipes"
                >
                  Recipes
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Categories
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Popular
                </Link>
              </h3>
              <div
                className="row"
              />
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="title"
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title__link"
                    replace={false}
                    to="/recipes"
                  >
                    Recipes
                  </Link>
                </h3>,
                <div
                  className="row"
                />,
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Categories
                  </Link>
                </h3>,
                <div
                  className="row"
                />,
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Popular
                  </Link>
                </h3>,
                <div
                  className="row"
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="title__link"
                    replace={false}
                    to="/recipes"
                  >
                    Recipes
                  </Link>,
                  "className": "title",
                  "style": Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Recipes",
                    "className": "title__link",
                    "replace": false,
                    "to": "/recipes",
                  },
                  "ref": null,
                  "rendered": "Recipes",
                  "type": [Function],
                },
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Categories
                  </Link>,
                  "style": Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Categories",
                    "className": "title-link",
                    "replace": false,
                    "to": "#",
                  },
                  "ref": null,
                  "rendered": "Categories",
                  "type": [Function],
                },
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Popular
                  </Link>,
                  "style": Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Popular",
                    "className": "title-link",
                    "replace": false,
                    "to": "#",
                  },
                  "ref": null,
                  "rendered": "Popular",
                  "type": [Function],
                },
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <section
            className="banner"
          >
            <div
              className="overlay"
            >
              <h3
                className="overlay__h3"
              >
                Welcome To Recipes. All About Reciping
              </h3>
            </div>
          </section>
          <section>
            <div
              className="container"
            >
              <h3
                className="title"
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title__link"
                  replace={false}
                  to="/recipes"
                >
                  Recipes
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Categories
                </Link>
              </h3>
              <div
                className="row"
              />
              <h3
                style={
                  Object {
                    "fontSize": 32,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="title-link"
                  replace={false}
                  to="#"
                >
                  Popular
                </Link>
              </h3>
              <div
                className="row"
              />
            </div>
          </section>
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="banner"
            >
              <div
                className="overlay"
              >
                <h3
                  className="overlay__h3"
                >
                  Welcome To Recipes. All About Reciping
                </h3>
              </div>
            </section>,
            <section>
              <div
                className="container"
              >
                <h3
                  className="title"
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title__link"
                    replace={false}
                    to="/recipes"
                  >
                    Recipes
                  </Link>
                </h3>
                <div
                  className="row"
                />
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Categories
                  </Link>
                </h3>
                <div
                  className="row"
                />
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Popular
                  </Link>
                </h3>
                <div
                  className="row"
                />
              </div>
            </section>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="overlay"
              >
                <h3
                  className="overlay__h3"
                >
                  Welcome To Recipes. All About Reciping
                </h3>
              </div>,
              "className": "banner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="overlay__h3"
                >
                  Welcome To Recipes. All About Reciping
                </h3>,
                "className": "overlay",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome To Recipes. All About Reciping",
                  "className": "overlay__h3",
                },
                "ref": null,
                "rendered": "Welcome To Recipes. All About Reciping",
                "type": "h3",
              },
              "type": "div",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <h3
                  className="title"
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title__link"
                    replace={false}
                    to="/recipes"
                  >
                    Recipes
                  </Link>
                </h3>
                <div
                  className="row"
                />
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Categories
                  </Link>
                </h3>
                <div
                  className="row"
                />
                <h3
                  style={
                    Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="title-link"
                    replace={false}
                    to="#"
                  >
                    Popular
                  </Link>
                </h3>
                <div
                  className="row"
                />
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="title"
                    style={
                      Object {
                        "fontSize": 32,
                        "marginTop": 40,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Link
                      className="title__link"
                      replace={false}
                      to="/recipes"
                    >
                      Recipes
                    </Link>
                  </h3>,
                  <div
                    className="row"
                  />,
                  <h3
                    style={
                      Object {
                        "fontSize": 32,
                        "marginTop": 40,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Link
                      className="title-link"
                      replace={false}
                      to="#"
                    >
                      Categories
                    </Link>
                  </h3>,
                  <div
                    className="row"
                  />,
                  <h3
                    style={
                      Object {
                        "fontSize": 32,
                        "marginTop": 40,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Link
                      className="title-link"
                      replace={false}
                      to="#"
                    >
                      Popular
                    </Link>
                  </h3>,
                  <div
                    className="row"
                  />,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="title__link"
                      replace={false}
                      to="/recipes"
                    >
                      Recipes
                    </Link>,
                    "className": "title",
                    "style": Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Recipes",
                      "className": "title__link",
                      "replace": false,
                      "to": "/recipes",
                    },
                    "ref": null,
                    "rendered": "Recipes",
                    "type": [Function],
                  },
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="title-link"
                      replace={false}
                      to="#"
                    >
                      Categories
                    </Link>,
                    "style": Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Categories",
                      "className": "title-link",
                      "replace": false,
                      "to": "#",
                    },
                    "ref": null,
                    "rendered": "Categories",
                    "type": [Function],
                  },
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="title-link"
                      replace={false}
                      to="#"
                    >
                      Popular
                    </Link>,
                    "style": Object {
                      "fontSize": 32,
                      "marginTop": 40,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Popular",
                      "className": "title-link",
                      "replace": false,
                      "to": "#",
                    },
                    "ref": null,
                    "rendered": "Popular",
                    "type": [Function],
                  },
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "section",
          },
        ],
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
